### PR TITLE
Add localization for the Ukrainian language in the calculator feature#419

### PR DESCRIPTION
**Pull request by lbeketl**
dev

* [Localization for the Ukrainian language in the calculator feature#419](https://github.com/ita-social-projects/ZeroWaste/issues/419)

## Code reviewers

- [ ] @loqimean

## Summary of change
* add  localization for the Ukrainian language in the calculator feature

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=95%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

image
image

### CODE REVIEW

File: app/models/calculator.rb [hunk 1]
```diff
@@ -29,7 +29,7 @@ class Calculator < ApplicationRecord
   accepts_nested_attributes_for :fields, allow_destroy: true
 
   validates :name, length: { minimum: 2 },
-                   format: { with: /\A[a-zA-Z0-9\s]+\z/, message: :name_format_validation },
+                   format: { with: /\A[a-zA-Zа-яА-Я0-9\s]+\z/, message: :name_format_validation },
                    uniqueness: true
```
Comment (AndriyAndriyovuch) [inline app/models/calculator.rb:33]: "ґрунт, підгузник, бур'ян, єнот, Їжак"

There names are not valid

**Review by AndriyAndriyovuch — CHANGES_REQUESTED** [review]

File: app/models/calculator.rb (hunk 1 above)
Comment (lbeketl) [reply app/models/calculator.rb:33]: resolved

File: spec/requests/calculators_spec.rb [hunk 2]
```diff
@@ -106,4 +106,32 @@
...
+        expect(flash[:notice]).to eq(I18n.t("notifications.calculator_created"))
+      end
+    end
+
+    context "with invalid attributes" do
+      it "does not create a calculator" do
+        expect do
+          post account_calculators_path, params: { calculator: invalid_attributes }
+        end.not_to change(Calculator, :count)
+
```
Comment (AndriyAndriyovuch) [inline spec/requests/calculators_spec.rb:132]: also add validation to error message you wrote in model's validation

**Review by AndriyAndriyovuch — CHANGES_REQUESTED** [review]

**Review by AndriyAndriyovuch — APPROVED** [review]

File: app/models/calculator.rb [hunk 3]
```diff
@@ -29,7 +29,7 @@ class Calculator < ApplicationRecord
   accepts_nested_attributes_for :fields, allow_destroy: true
 
   validates :name, length: { minimum: 2 },
-                   format: { with: /\A[a-zA-Z0-9\s]+\z/, message: :name_format_validation },
+                   format: { with: /\A[a-zA-Zа-яієїґ'А-ЯІЄЇҐ0-9\s]+\z/, message: :name_format_validation },
```
Comment (loqimean) [inline app/models/calculator.rb:32]: а навіщо тут кавичка?

File: spec/requests/calculators_spec.rb [hunk 4]
```diff
@@ -106,4 +106,33 @@
       end
     end
   end
+
+  describe "POST /account/calculators" do
```
Comment (loqimean) [inline spec/requests/calculators_spec.rb:110]: describe "METHOD #action"

File: app/models/calculator.rb (hunk 3 above)
Comment (lbeketl) [reply app/models/calculator.rb:32]: Це апостроф, типу якшо хтось схоче назвати калькулятор словом в якому буде '

Comment (loqimean) [reply app/models/calculator.rb:32]: а, окі, ну насправді я б не робив цієї валідації.., але нехай вже буде

**Review by loqimean — APPROVED** [review]

**Pull request merged**